library/n2_protocol.hpp

```diff
@@ -8,14 +8,35 @@
 
 namespace ioremap { namespace elliptics { namespace n2 {
 
-struct data_in_file {
-	int fd = -1;
-	off_t local_offset = 0;
-	size_t fsize = 0;
-	int on_exit = 0;
+class data_in_file {
+public:
+	struct info_t {
+		int fd = -1;
+		off_t local_offset = 0;
+		size_t fsize = 0;
+	};
+
+	data_in_file() = default;
+	data_in_file(data_in_file &&) = default;
+	data_in_file &operator = (data_in_file &&) = default;
+
+	~data_in_file();
+
+	const info_t &info() const;
+
+	void detach();
+	data_pointer read_and_detach();
+
+private:
+	data_in_file(const data_in_file &) = delete;
+	data_in_file &operator = (const data_in_file &) = delete;
+
+	info_t info_;
+	int on_exit_ = 0;
 };
 
-struct data_place {
+class data_place {
+public:
```

Comment:
I'm not sure, but in class all members must be in private section, mustn't they?

Reply:
In google code style must be. I rework this class, but you'll see, it'll express to big overhead.